QRCoder/QRCodeGenerator.ModulePlacer.BlockedModules.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System;
+using System.Collections;
+
+namespace QRCoder
+{
+    public partial class QRCodeGenerator
+    {
+        private static partial class ModulePlacer
+        {
+            /// <summary>
+            /// Struct that represents blocked modules using rectangles.
+            /// </summary>
+            public struct BlockedModules : IDisposable
+            {
+                private readonly bool[,] _blockedModules;
```

Comment:
Rectangular arrays have quite bad perf compared to jagged arrays (`T[][]`), because the latter are so called "single dimensional zero indexed arrays" (sz-arrays) and have betterr support within the JIT and runtime.

To make the code easier, I'd create a matrix type that is based on the sz-array, and have an indexer that makes the work with the rows/columns easier.

Reply:
Changed to `bool[][]`

Reply:
Also, using a `BitArray[]` is another choice.

Reply:
Using `BitArray[]` is perhaps 5% slower for tiny QR codes (just 5 us), but negligible for longer QR codes.  We use `BitArray` elsewhere in the codebase so perhaps it makes sense to use it here.

Reply:
Based on your 👍 above, I've changed to `BitArray[]`, which is more similar to the rest of the codebase.